superset/utils/date_parser.py

```diff
@@ -71,19 +71,32 @@ def parse_human_datetime(human_readable: str) -> datetime:
     >>> year_after_1 == year_after_2
     True
     """
+    x_periods = r"^\s*([0-9]+)\s+(second|minute|hour|day|week|month|quarter|year)s?\s*$"
+    if re.search(x_periods, human_readable, re.IGNORECASE):
+        raise ValueError(
+            _(
+                "date string is unclear."
+                " Please specify [{0} ago] or [{0} later]".format(human_readable)
+            )
+        )
+
+    error_msg = ValueError(_("Couldn't parse date string [{}]".format(human_readable)))
```

Comment:
Proper i18n should use the built-in interpolation from `_`:

```python
_("Couldn't parse date string [%(input)s]", input="human_readable")
```

Reply:
Thanks for pointing this out. I remember that `lazy_gettext()` function can support this syntax, but I've tried this and now it doesn't work.  I will fix this.